solutions/usaco-737.mdx

```diff
@@ -12,38 +12,169 @@ author: Juheon Rhee
 **Time Complexity**: $\mathcal O(N^3)$
 
 <LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <fstream>
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+const int MAX_COLOR = 9;
+
+int main() {
+	// The first element won't be used (colors go from 1-9)
+	vector<int> left(MAX_COLOR + 1);
+	vector<int> right(MAX_COLOR + 1);
+	vector<int> up(MAX_COLOR + 1);
+	vector<int> down(MAX_COLOR + 1);
+	for (int c = 1; c <= MAX_COLOR; c++) {
+		left[c] = up[c] = INT32_MAX;
+		right[c] = down[c] = -1;
+	}
+
+	vector<bool> valid_start(MAX_COLOR + 1);
+	std::ifstream read("art.in");
+	int width;
+	read >> width;
+	vector<vector<int>> art(width, vector<int>(width));
+	for (int r = 0; r < width; r++) {
+		for (int c = 0; c < width; c++) {
+			char curr_char;
+			read >> curr_char;
+			int curr = curr_char - '0';
+			art[r][c] = curr;
+			if (curr) {
+				left[curr] = std::min(left[curr], c);
+				right[curr] = std::max(right[curr], c);
+				down[curr] = std::max(down[curr], r);
+				up[curr] = std::min(up[curr], r);
+				valid_start[curr] = true;
+			}
+		}
+	}
+
+	for (int col = 1; col <= MAX_COLOR; col++) {
```

Comment:
i think you should spell out the words "color" or "column", since they both start with "col" lol